public/app/features/alerting/unified/mockApi.ts

```diff
@@ -424,10 +424,10 @@ export function mockDashboardApi(server: SetupServer) {
   };
 }
 
+const server = setupServer();
```

Comment:
We can set up the server every time regardless, as it's the act of `.listen` etc. that takes effect in the tests

Reply:
I'm usually not a big fan of singletons, but I can see that this does simplify the API so we don't have to explicitly pass the server instance in to all of the higher level functions to overwrite the handler responses.

Perhaps this function could return an interface that implements the higher level functions defined in `configure.ts`? ie.

```tsx
export function setupMswServer () {
   const server = setupServer();
 
   return {
       setAlertmanagerChoices: (...args) => setAlertmanagerChoices(server, ...args),
   }
}
```

and consume as such

```tsx
  const server = setupMswServer();
  server.setAlertmanagerChoices(AlertmanagerChoice.Internal, 1);
```

Reply:
I'm not sure I'd want to tie the two things together in that particular way. I was thinking that `setupServer` would only be concerned with sorting the jest before/after lifecycle hooks, so we could one day remove it. What about if the pattern for the config methods was more like this, with ability to chain methods?

```ts
configureServer
  .setAlertmanagerChoices(...)
  .doSomethingElse(...)
```

Reply:
Yea not conflating the two is reasonable, my point was avoiding the singleton because in the past this has caused weird issues and race conditions when running tests in parallel that are updating handlers on the same instance.